Gui/QGLWidgetCompat.h

```diff
@@ -0,0 +1,28 @@
+/* ***** BEGIN LICENSE BLOCK *****
+ * This file is part of Natron <https://natrongithub.github.io/>,
+ * (C) 2018-2022 The Natron developers
+ * (C) 2013-2018 INRIA and Alexandre Gauthier-Foichat
+ *
+ * Natron is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Natron is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Natron.  If not, see <http://www.gnu.org/licenses/gpl-2.0.html>
+ * ***** END LICENSE BLOCK ***** */
+
+#ifndef NATRON_GUI_QGLWIDGETCOMPAT_H
+#define NATRON_GUI_QGLWIDGETCOMPAT_H
+
+#include <QGLWidget>
```

Comment:
- add a comment at the top explaining what this is
- make sure GLIncludes.h is included before
- use the full include path `<QtOpenGL/QGLWidget>`
- OpenGL is deprecetad under macOS, silence it (see below)
- check that this is not included in Qt >= 5.4
```
#include "Global/Macros.h"
#include "Global/GLIncludes.h" //!<must be included before QGlWidget because of gl.h and glew.h
CLANG_DIAG_OFF(deprecated)
#include <QtOpenGL/QGLWidget>
CLANG_DIAG_ON(deprecated)
#if QT_VERSION >= QT_VERSION_CHECK(5, 4, 0)
#error "this file must not be included in Qt >= 5.4.0"
#endif
```

Reply:
I went with checking that `GLIncludes.h` was before the `QGL*Compat.h` headers in the sources.